File: packages/transform/src/features/protobuf/ProtobufEncodeTransformer.ts
import { ProtobufEncodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.encode",
      write: ProtobufEncodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufEncodeTransformer = protobufEncodeTransformerNamespace{}

type protobufEncodeTransformerNamespace struct{}

func (protobufEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufEncodeProgrammer.Write(call.TypeText)
}
